### PR TITLE
fix mac build

### PR DESCRIPTION
**Pull request by yiwu-arbug**
It appears on MacOS with Clang, `bool` is not supported in C, causing compile error. Changing it to `char`.

Signed-off-by: Yi Wu <yiwu@pingcap.com>

### CODE REVIEW

**Review by tabokie — APPROVED** [review]
Or unsigned char?

**Comment by MrCroxx** [comment]
> Or unsigned char?

IMO char is fine, like the most libs.

**Review by MrCroxx — APPROVED** [review]

**Comment by tabokie** [comment]
> > Or unsigned char?
> 
> IMO char is fine, like the most libs.

"most libs" is not relevant here. What's relevant is consistency within "this lib".

**Comment by yiwu-arbug** [comment]
Updated all boolean in crocksdb into unsigned char.

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -524,15 +524,18 @@ struct crocksdb_mergeoperator_t : public MergeOperator {
...
-  char* (*full_merge_)(void*, const char* key, size_t key_length,
-                       const char* existing_value, size_t existing_value_length,
-                       const char* const* operands_list,
-                       const size_t* operands_list_length, int num_operands,
-                       unsigned char* success, size_t* new_value_length);
-  char* (*partial_merge_)(void*, const char* key, size_t key_length,
-                          const char* const* operands_list,
-                          const size_t* operands_list_length, int num_operands,
-                          unsigned char* success, size_t* new_value_length);
+  unsigned char* (*full_merge_)(void*, const char* key, size_t key_length,
```
Comment (tabokie) [inline librocksdb_sys/crocksdb/c.cc:527]: This should be left alone. It returns a new value not a boolean pointer.

Comment (yiwu-arbug) [reply librocksdb_sys/crocksdb/c.cc:527]: ah, good catch. I misread it.

**Pull request merged**